packages/utils-dom/src/htmldocument.ts

```diff
@@ -165,8 +165,17 @@ export class HTMLDocument extends Node {
     /**
      * https://developer.mozilla.org/docs/Web/API/Document/querySelector
      */
-    public querySelector(selector: string): HTMLElement {
-        return this.querySelectorAll(selector)[0];
+    public querySelector(selector: string): HTMLElement | null {
+        if (!CACHED_CSS_SELECTORS.has(selector)) {
+            CACHED_CSS_SELECTORS.set(selector, cssSelect.compile(selector));
+        }
+
+        const data: any | undefined = cssSelect.selectOne(
```

Comment:
Can we use `NodeData` in here or is that causing any types issue? If so, maybe we should cast it already here instead of pass `any` around?

Reply:
Good point, let me give it a try.

Reply:
I resolved this by removing the explicit. selectOne returns a complex union type that is filtered down by the arguments, in this case it returns `ChildData` which is compatible with `NodeData`.

By setting the typing to be implicit we let the TyepScript compiler do the work for us :)